website_sale_product_attribute_value_filter_existing/controllers/main.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2019 Tecnativa - Sergio Teruel
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class ProductAttributeValues(WebsiteSale):
+    def _get_search_domain(
+        self, search, category, attrib_values, search_in_description=True
+    ):
+        # Store used domain in context to be reused after
+        domain = super()._get_search_domain(
+            search, category, attrib_values, search_in_description=search_in_description
+        )
+        new_context = dict(request.env.context, shop_search_domain=domain)
+        request.update_context(**new_context)
+        return domain
+
+    @http.route()
+    def shop(self, page=0, category=None, search="", ppg=False, **post):
+        res = super().shop(page=page, category=category, search=search, ppg=ppg, **post)
+        domain = request.env.context.get("shop_search_domain", [])
+        # Load all products without limit for the filter check on
+        # attribute values
+        templates = request.env["product.template"].search(domain, limit=False)
+        ProductTemplateAttributeLine = request.env["product.template.attribute.line"]
+        attribute_values = ProductTemplateAttributeLine.search(
+            [("product_tmpl_id", "in", templates.ids)]
+        )
```

Comment:
@anothingguy better to do lazy:

```
attribute_values = lazy(ProductTemplateAttributeLine.search(
            [("product_tmpl_id", "in", templates.ids)]
))
```

Reply:
lazy won't be useful here because attribute_values is called only one time, so I keep it as-is in v15

Reply:
we could get the benefit of `lazy` in the case that other modules inherit this one, then change the way to get attribute_values. 
here is an example we have in `website_sale`:
```
            # get all products without limit
            attributes = lazy(lambda: ProductAttribute.search([
                ('product_tmpl_ids', 'in', search_product.ids),
                ('visibility', '=', 'visible'),
            ]))
```

Reply:
`attributes` in `website_sale` is returned in results and used in template, so it may be used many times and `lazy` is useful in that case. `attribute_values` is not that case in this module. So I want to keep minimal changes if needed. But sure, if we use `lazy` here, it's still fine but I don't see any benefit actually

Reply:
updated as well

Reply:
>attribute_values = lazy(`lambda`: ...)

you need to call your function inside a `lambda`. Thank to that, your funcion `ProductTemplateAttributeLine.search()` will only execute when `attribute_values` is used.